File: Backend/src/models/EmergencyAction.js
const mongoose = require('mongoose');

const EmergencyActionSchema = new mongoose.Schema(
  {
    journeyId: { type: mongoose.Schema.Types.ObjectId, ref: 'Journey', required: true },
    userId: { type: mongoose.Schema.Types.ObjectId, ref: 'User', required: true },
    timestamp: { type: Date, required: true },
    action: {
      type: String,
      enum: ['sos_call', 'voice_recording', 'no_response'],
      required: true,
    },
    audioUrl: String,
    notifiedParents: [{ type: mongoose.Schema.Types.ObjectId, ref: 'User' }],
  },
  { timestamps: true }
);

module.exports = mongoose.model('EmergencyAction', EmergencyActionSchema);

File: women_safety_user/lib/src/models/emergency_action.dart
class EmergencyAction {
  final String id;
  final String journeyId;
  final String userId;
  final String action;
  final String audioUrl;

  EmergencyAction({required this.id, required this.journeyId, required this.userId, required this.action, required this.audioUrl});

  factory EmergencyAction.fromJson(Map<String, dynamic> json) => EmergencyAction(
    id: json['_id'] ?? '',
    journeyId: json['journeyId'],
    userId: json['userId'],
    action: json['action'],
    audioUrl: json['audioUrl'] ?? '',
  );
}
